tinygrad/tensor.py

```diff
@@ -481,6 +481,12 @@ def relu(self): return mlops.Relu.apply(self)
   def sin(self): return mlops.Sin.apply(self)
   def cos(self): return ((math.pi/2)-self).sin()
   def tan(self): return self.sin() / self.cos()
+  
+  @staticmethod
+  def _tri(r:int, c:int, k:int=0) -> Tensor: return Tensor.arange(r).unsqueeze(1).expand(r,c) <= Tensor.arange(c-k, start=-k).unsqueeze(0).expand(r,c)
+  def triu(self, k:int=0) -> Tensor: return Tensor._tri(self.shape[0], self.shape[1], k=k).where(self, Tensor.zeros_like(self))
+  def tril(self, k:int=0) -> Tensor: return Tensor._tri(self.shape[0], self.shape[1], k=k+1).where(Tensor.zeros_like(self), self)
```

Comment:
Shouldn't these be self.shape[-2] and self.shape[-1] to match numpy/torch? https://numpy.org/doc/stable/reference/generated/numpy.triu.html

Reply:
will change to match, couldn't use the unpack due to linter